### PR TITLE
Add spec hook for replacing old version string

### LINKED ISSUE #382: RFE: spec hook for replacing occurences of old version string in a spec file
Example:

```specfile
Summary: Timezone data
Name: tzdata
Version: 2018c
%define tzdata_version 2018c
%define tzcode_version 2018c
Release: 2%{?dist}
License: Public Domain
Group: System Environment/Base
URL: https://www.iana.org/time-zones
Source0: ftp://ftp.iana.org/tz/releases/tzdata%{tzdata_version}.tar.gz
Source1: ftp://ftp.iana.org/tz/releases/tzcode%{tzcode_version}.tar.gz

...
```

### PR DESCRIPTION
**Pull request by FrNecas**
Resolves #382

### CODE REVIEW

File: rebasehelper/spec_hooks/replace_old_version.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                be used as a substitution.
+
+        Returns:
+            str: Modified line with replaced old version strings.
+
+        """
+        if re.search(r"Version:", line):
+            return line
+
+        comment = re.search(r"#.*", line)
```
Comment (nforro) [inline rebasehelper/spec_hooks/replace_old_version.py]: Is there a reason to use regular expressions here at all? What about this?

```python
try:
    comment = line.index('#')
except ValueError:
    comment = None
return line.replace(old, '%{version}' if replace_with_macro else new, line[:comment]) + line[comment:]
```

Comment (FrNecas) [reply rebasehelper/spec_hooks/replace_old_version.py]: What about setting comment to len(line) - 1, if no `#` is found? This way, the condition regarding comment wouldn't be necessary

Comment (nforro) [reply rebasehelper/spec_hooks/replace_old_version.py]: Good idea, but why `- 1`?

Comment (FrNecas) [reply rebasehelper/spec_hooks/replace_old_version.py]: Oh, `-1` shouldn't be there, it is a slice.

File: rebasehelper/spec_hooks/replace_old_version.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            old (str): Old version string.
+            new (str): New version string.
+            replace_with_macro (bool): Whether %{version} macro should
+                be used as a substitution.
+
+        Returns:
+            str: Modified line with replaced old version strings.
+
+        """
+        if re.search(r"Version:", line):
```
Comment (nforro) [inline rebasehelper/spec_hooks/replace_old_version.py]: Is this check really necessary? Isn't the version already bumped in `rebase_spec_file`?

**Pull request merged**